src/libs/kata-types/src/capablities.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use bitmask_enum::bitmask;
+
+#[bitmask(u8)]
+enum CapabilityBits {
+    blockDeviceSupport,
+    blockDeviceHotplugSupport,
+    multiQueueSupport,
+    fsSharingSupported,
+}
+
+/// Capabilities describe a virtcontainers hypervisor capabilities through a bit mask.
+#[derive(Debug, Clone)]
+pub struct Capabilities {
+    /// flags
+    flags: CapabilityBits,
+}
+
+impl Default for Capabilities {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Capabilities {
+    /// new Capabilities struct
+    pub fn new() -> Self {
+        Capabilities {
+            flags: CapabilityBits { bits: 0 },
+        }
+    }
+
+    /// is_block_device_supported tells if an hypervisor supports block devices.
```

Comment:
Considering future extensions, these `is_xxx` and `set_xxx` can be generated by a macro.
But you can leave it in another PR.